Assets/Scripts/Models/Buildable/Components/PowerConnection.cs

```diff
@@ -0,0 +1,208 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml.Serialization;
+using UnityEngine;
+
+namespace ProjectPorcupine.Buildable.Components
+{
+    [Serializable]
+    [XmlRoot("Component")]
+    [BuildableComponentName("PowerConnection")]
+    public class PowerConnection : BuildableComponent
+    {
+        // constants for parameters
+        public const string CurAccumulatorChargeParamName = "pow_accumulator_charge";
+        public const string CurAccumulatorIndexParamName = "pow_accumulator_index";
+        public const string CurIsRunningParamName = "pow_is_running";
+        
+        public PowerConnection()
+        {
+        }
+
+        private PowerConnection(PowerConnection other) : base(other)
+        {
+            Provides = other.Provides;
+            Requires = other.Requires;
+        }
+              
+        [XmlIgnore]
+        public bool IsRunning
+        {
+            get
+            {
+                bool running = false;
+
+                // TODO: make adjustments in Parameters class to prevent double check for ContainsKey (indexer already checks it)
```

Comment:
Please don't make that adjustment in Parameters, it is an intentional design that Parameters autocreates any parameter that is attempted to be accessed, and that requires it to check if it contains the key.